### PR TITLE
Semantic versioning npm

### PR DESCRIPTION
**Pull request by bjmccotter7192**
## Description
In this PR I have updated the github actions to use semantic-release. This change will allow contributors to control the version of their package with specific commit message prefixes.  

Semantic-release also creates a Github release and tag that is matched with the NPM package version. Once the release and tag are created it publishes the package to [@department-of-veterans-affairs/va-forms-system-core](https://www.npmjs.com/package/@department-of-veterans-affairs/va-forms-system-core).

## Links
- [Semantic-Release](https://github.com/semantic-release/semantic-release)

## Quick How to use semantic-release
- Prefix your commit messages:

Commit message | Release type
-- | --
fix(pencil): stop graphite breaking (ex. 0.0.1) | Patch Fix Release
feat(pencil): add 'graphiteWidth' option (ex. 0.1.0)  | Minor Feature Release
perf(pencil): remove graphiteWidth option BREAKING CHANGE: The graphiteWidth option has been removed. The default graphite width of 10mm is always used for performance reasons. (ex. 1.0.0) | Major Breaking Release(Note that the BREAKING CHANGE: token must be in the footer of the commit)

## Original issue(s)
[department-of-veterans-affairs/va-forms-system-core#432](https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/432)

## Testing done
- [ ] All Unit Tests Are Passing

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by micahchiang — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -5,6 +5,43 @@
...
+            }
+          ],
+          "parserOpts": {
+            "noteKeywords": [
+              "BREAKING CHANGE",
+              "BREAKING CHANGES"
+            ]
+          }
+        }
+      ],
```
Comment (nickjg231) [inline package.json:38]: This is pretty cool. Will this require all of our commits (or at least 1 commit per PR) to use this angular convention?

Comment (bjmccotter7192) [reply package.json:38]: Exactly so it goes through the commits and analyzes the messages, looking for (fix, feat, or perf "With perf you have to use the BREAKING CHANGES: at the footer of the commit message).

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -36,8 +36,15 @@ jobs:
...
         env_variable_name: VA_VSP_BOT_NPM_TOKEN
-    
-    - name: Publish build artifacts to NPM
-      run: yarn npm publish --access public --tolerate-republish
+
+    - name: Get Github Token password
+      uses: marvinpinto/action-inject-ssm-secrets@v1.2.1
+      with:
+        ssm_parameter: /devops/VA_VSP_BOT_GITHUB_TOKEN
+        env_variable_name: VA_VSP_BOT_GITHUB_TOKEN
```
Comment (nickjg231) [inline .github/workflows/npm-publish.yml:44]: Where do these variables come from?

**Review by nickjg231 — APPROVED** [review]
Looks good to me, just curious how a few things work 👨‍🏫

File: .github/workflows/npm-publish.yml (hunk 2 above)
Comment (bjmccotter7192) [reply .github/workflows/npm-publish.yml:44]: They are pulled from the secret store in AWS so you use the AWS login action above and then you have access to those secrets

**Comment by bjmccotter7192** [comment]
@department-of-veterans-affairs/vsp-operations Could someone check and make sure that our repo has access to the VA_VSP_BOT_GITHUB_TOKEN so that the semantic-release/git and semantic-release/github actions can properly create Github Releases, tags, and commits. Thanks!

**Comment by jsgarmon** [comment]
currently this service account only has access to `VA_VSP_BOT_NPM_TOKEN`. I'll put in a PR to have the github token added to its perms.

**Comment by bjmccotter7192** [comment]
@jsgarmon Thank you so much!

**Review by jsgarmon — APPROVED** [review]
good to go

**Pull request merged**